BinaryOperations/README.md

```diff
@@ -0,0 +1,77 @@
+# How do I fold over two structures at once?
+
+There are many operations where we want to look at two structures. The first one that comes to mind is often equality:
+```haskell
+equal :: (a, a) -> Bool
+```
+This is something we end up implementing all the time (or often deriving), but when we’re working with recursion schemes, how can we do this with an algebra?
+
+I’ll spoil the trick up-front: We write an algebra over the _first_ parameter, where the [carrier](../glossary.md#carrier) is a function from the second parameter to the result, like so
+```haskell
+equal' :: Projectable t f => f (t -> Bool)
+
+cata equal' :: (Recursive t f, Projectable t f) => t -> t -> Bool
+```
+
+The type on `cata equal'` there applies the fold to the _first_ `t`, and the resulting `t -> Bool`  makes it look exactly like the `equal` operation we want.
+
+This shape of algebra is an [attribute grammar](../AttributeGrammars/README.md). But this specific case is for when both parameters are recursive structures that we want to align from the _root_.
```

Comment:
It'd be informative to see worked examples to understand this further. So like comparing two `Nat`s, or `Tree`s, or `List`s, etc.